shared/middleware/tables_api.rb

```diff
@@ -44,6 +44,31 @@ class TablesApi < Sinatra::Base
   end
 
   #
+  # post /v3/(shared|user)-tables/<channel-id>/<table-name>/metadata
+  #
+  # Sets the metdata for the given table
+  #
+  post %r{/v3/(shared|user)-tables/([^/]+)/([^/]+)/metadata$} do |endpoint, channel_id, table_name|
```

Comment:
nit: REST says this should be a PUT, even though I know we haven't been consistent about this, since we are setting the value at this URI rather than adding to it as a collection.